staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -458,6 +458,18 @@ func (c *AvailableConditionController) updateAPIServiceStatus(originalAPIService
 		return newAPIService, nil
 	}
 
+	orig := apiregistrationv1apihelper.GetAPIServiceConditionByType(originalAPIService, apiregistrationv1.Available)
+	now := apiregistrationv1apihelper.GetAPIServiceConditionByType(newAPIService, apiregistrationv1.Available)
+	if orig == nil {
+		orig = &apiregistrationv1.APIServiceCondition{
+			Type:   apiregistrationv1.Available,
+			Status: apiregistrationv1.ConditionUnknown,
+		}
+	}
+	if *orig != *now {
+		klog.V(2).InfoS("changing APIService availability", "name", newAPIService.Name, "oldStatus", orig.Status, "newStatus", now.Status, "message", now.Message, "reason", now.Reason)
```

Comment:
Seems like most of the add/update logs are V(4). Do we want this one at V(2)?

Reply:
This is so important for cluster/API availability, I think `V(2)` makes super sense here. When this flaps, it is critical for the cluster.

Reply:
I would even go one step further and create events (in an follow-up) if we don't have that yet.